dds/DCPS/transport/framework/TransportRegistry.cpp

```diff
@@ -71,20 +64,35 @@ const OPENDDS_STRING TransportRegistry::CUSTOM_ADD_DOMAIN_TO_PORT = "add_domain_
 TransportRegistry::TransportRegistry()
   : global_config_(make_rch<TransportConfig>(DEFAULT_CONFIG_NAME))
   , released_(false)
-  , load_complete_(lock_)
 {
   DBG_ENTRY_LVL("TransportRegistry", "TransportRegistry", 6);
   config_map_[DEFAULT_CONFIG_NAME] = global_config_;
+}
 
-  lib_directive_map_["tcp"]       = "dynamic OpenDDS_Tcp Service_Object * OpenDDS_Tcp:_make_TcpLoader()";
-  lib_directive_map_["udp"]       = "dynamic OpenDDS_Udp Service_Object * OpenDDS_Udp:_make_UdpLoader()";
-  lib_directive_map_["multicast"] = "dynamic OpenDDS_Multicast Service_Object * OpenDDS_Multicast:_make_MulticastLoader()";
-  lib_directive_map_["rtps_udp"]  = "dynamic OpenDDS_Rtps_Udp Service_Object * OpenDDS_Rtps_Udp:_make_RtpsUdpLoader()";
-  lib_directive_map_["shmem"]     = "dynamic OpenDDS_Shmem Service_Object * OpenDDS_Shmem:_make_ShmemLoader()";
+namespace {
+void load_dynamic_library(const char* directive)
+{
+  const ACE_TString d = ACE_TEXT_CHAR_TO_TCHAR(directive);
+  if (ACE_Service_Config::process_directive(d.c_str()) != 0) {
+    ACE_DEBUG((LM_NOTICE,
```

Comment:
Shouldn't this be either an [error or info since we can't give feedback to the user](https://opendds.readthedocs.io/en/master/internal/dev_guidelines.html#logging-conditions-and-priority)?

Reply:
It's not an error if they chose to not build the library.

Reply:
That's true, but in the guidelines we decided to pair `LM_NOTICE` with `ACE_ERROR` and give it a specific meaning as "reportable error".  So maybe this an info level message?